### ISSUE TITLE
Cell Basilica block monster attack but didn't block monster entering/use skill.

### ISSUE THREAD

**Issue by Emistry** [HerculesWS/Hercules#2420]
**Describe the bug**
Monster that entered basilica area doesn't chase player who left the basilica area.

**To Reproduce**
1. Load this npc script
```
map,45,45,4	script	Sample	4_F_KAFRA1,{
	end;
	
	OnInit:
		getmapxy(.@map$, .@x, .@y, UNITTYPE_NPC);
		.@range = 3;
		setcell(.@map$, (.@x - .@range), (.@y - .@range), (.@x + .@range), (.@y + .@range), cell_basilica, true);
		end;
}
```
2. summon any aggresive monsters _(I tried with MVP monster)_
3. allow the monster to chase the character and enter the basilica area
4. move the character out from basilica area

**Expected behavior**
1. ~Monster shall be pushed away or refrain from entering basilica area. _(not sure if MVP should be refrain from entering too, skill description seem allow it)_~ [Any non-boss monsters in a 5 X 5 area next to the caster when Basilica's cast time is complete will be instantly pushed out of the area.](https://irowiki.org/wiki/Basilica)
2. If monster currently within the basilica area, it shall chase any characters if the they aren't within basilica area (within monster seen/chase/attack range).
3. ~Monster shouldn't able to cast/place AOE skills within the basilica area, although it doesn't deal any damage.~ [AOE Skills seem like still possible, Ex: Venom Splasher](https://irowiki.org/wiki/Basilica)

**Screenshots**
![pic](https://i.imgur.com/6F07PsB.png)

**System specs:**
 - OS:  window 10
 - Hercules Version https://github.com/HerculesWS/Hercules/commit/e1e951c805916853e55ff5b9ce0531e0cf483ebf
 - Mode: renewal
 - Packet version: 20180620
 - Client type: RE

**Plugins used or source modifications**
None

**Additional context**
Probably related Issues : https://github.com/HerculesWS/Hercules/issues/789 and https://github.com/HerculesWS/Hercules/issues/1991

Not sure if this is the actual behavior that currently work in kRO.

**Comment by lllaaazzz** [comment]
Like this  ? 
https://im3.ezgif.com/tmp/ezgif-3-e05652450914.gif

**Comment by Emistry** [comment]
what I trying to said in my case:
1. a MVP is chasing player
2. player enter the basilica, MVP will still chasing me and entered the basilica area with me
3. when the MVP entered basilica, it just randomly walk around the basilica area.
4. If a player now walk outside basilica area, but right in front of the MVP (MVP still inside basilica area but player aren't), the MVP doesn't chase that player.
5. If i walk out from basilica area and stand in front MVP, the MVP doesnt chase me too. (MVP inside basilica area, but player aren't)
6. Then if the MVP now walk to outside the basilica (both outside basilica area), it will start chase/attacking again.

the MVP become something like idle state when it entered the basilica area. It just walk randomly in the basilica area until it left the basilica area, only then the MVP will back to chase/attack player again.

**Comment by lllaaazzz** [comment]
I dont know the official mechanics but when i do this with an mvp, he just stands outside the bascila zone, and itll chase the first thing it sees outside of it, minions arents effected by knockback .... and he cant enter the zone at all

**Comment by lllaaazzz** [comment]
why not ask rytech or dastgir

**Comment by skyleo** [comment]
MVPs interaction with Basilica is bugged until this is merged:
https://github.com/HerculesWS/Hercules/pull/2612

**Comment by Jedzkie** [comment]
Closing this issue, fixed in #2612
